graphql_compiler/schema_transformation/rename_query.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+from copy import deepcopy
+
+from graphql.language import ast as ast_types
+from graphql.language.visitor import Visitor, visit
+
+from .utils import QueryStructureError
+
+
+def rename_query(ast, renamings):
+    """Translate all types and root fields (fields of query type) using renamings.
+
+    Most fields (fields of types other than the query type) will not be renamed. Any type
+    or field that does not appear in renamings will be unchanged.
+    Roughly speaking, rename_query renames the same elements as rename_schema.
+
+    Args:
+        ast: Document, representing a valid query. It is assumed to have passed GraphQL's
+             builtin validation -- validate(schema, ast). ast not modified by this function
+        renamings: Dict[str, str], mapping original types/query type field names as appearing
+                   in the query to renamed names. Type or query type field names not appearing
+                   in the dict will be unchanged
+
+    Returns:
+        Document, a new AST representing the renamed query
+
+    Raises:
+        - QueryStrutureError if the ast does not have the expected form; in particular, if the
+          AST contains Fragments, or if it contains an InlineFragment at the root level
+    """
+    # NOTE: There is a validation section in graphql-core that takes in a schema and a
+    # query ast, and checks whether the query is valid. This code assumes this validation
```

Comment:
Instead of mentioning the validation functions here and in the docstring, I would mention what they validate instead.

Reply:
The full list of validations is long (specified_rules  in https://github.com/graphql-python/graphql-core/blob/05a4c8222c0b65ff8df61564fe7f41efee060b4e/graphql/validation/rules/__init__.py), I'll mention a few of them as examples.